src/citrine/_serialization/properties.py

```diff
@@ -101,40 +135,54 @@ def deserialize_from_dict(self, data: dict) -> DeserializedType:
                 break
             else:
                 value = next_value
-        base_class = _get_base_class(data, self.serialization_path)
-        return self.deserialize(value, base_class=base_class)
+        return self.deserialize(value, base_class=None)  # always a dict
 
     def serialize_to_dict(self, data: dict, value: DeserializedType) -> dict:
         if self.serialization_path is None:
             raise ValueError('No serialization path set!')
-        else:
-            base_class = _get_base_class(data, self.serialization_path)
```

Comment:
Why is resolving the base class no longer necessary along this code path?

Reply:
https://github.com/CitrineInformatics/citrine-python/blob/9be3cded48ea64f754d1897513100fd2230d0c52/src/citrine/_serialization/properties.py#L666-L686

There's no way to get to that line where `data` is not a `dict`.